python/ray/air/_internal/torch_utils.py

```diff
@@ -168,11 +171,16 @@ def load_torch_model(
 ) -> torch.nn.Module:
     """Loads a PyTorch model from the provided ``saved_model``.
 
-    If ``saved_model`` is a torch Module, then return it directly. If ``saved_model`` is
-    a torch state dict, then load it in the ``model_definition`` and return the loaded
-    model.
+    ``model_definition`` is used and only used when ``saved_model`` is
+    a torch state dict, which will be loaded into ``model_definition``.
+    Otherwise, ``model_definition`` is discarded.
     """
     if isinstance(saved_model, torch.nn.Module):
+        if model_definition:
+            logger.warning(
+                "TorchCheckpoint already contains all information needed. "
+                "Discarding provided `model_definition`. "
+            )
```

Comment:
Could we move this warning to `TorchCheckpoint`? `TorchCheckpoint.from_model` doesn't have a `model_definition` parameter, so this warning might be confusing to the user.

```
def from_model(
        cls,
        model: torch.nn.Module,
        *,
        preprocessor: Optional["Preprocessor"] = None,
    )
```

Reply:
Ah good point. Do you mean TorchPredictor and BatchPredictor right? (not TorchCheckpoint.from_model)
I still think this is a good place to throw the error just making it more explicit/actionable, as user may be calling either TorchPredictor.from_checkpoint or BatchPredictor.from_checkpoint(..., TorchPredictor,...)
Updated the error msg. PTAL.

Reply:
Ah, yeah, you're right. The change makes more sense for `TorchPredictor` and `BatchPredictor`.

Both `TorchPredictor.from_checkpoint` and `BatchPredictor.from_checkpoint` eventually call `TorchCheckpoint.get_model`, so we could put the warning there if we want to raise the warning earlier. I'm okay either way.